src/sentry/utils/kafka_config.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import
+from django.conf import settings
+
+
+def _get_legacy_kafka_cluster_options(cluster_name):
+    options = settings.KAFKA_CLUSTERS[cluster_name]
+
+    return {k: v for k, v in options.items() if k not in ("common", "producers", "consumers")}
+
+
+def _get_kafka_cluster_options(cluster_name, config_section, with_legacy=False, override=None):
+    options = {}
+    if override:
+        options.update(override)
+    if with_legacy:
+        legacy_options = _get_legacy_kafka_cluster_options(cluster_name)
+        options.update(legacy_options)
+    custom_options = settings.KAFKA_CLUSTERS[cluster_name].get(config_section, {})
+    common_options = settings.KAFKA_CLUSTERS[cluster_name].get("common", {})
+    options.update(custom_options)
+    options.update(common_options)
```

Comment:
I think the order these parameters are being applied is not the right one.
I would say it should be:
- common_options
- custom_options
OR 
- legacy options (if present)
THEN override_options.

So legacy options replace the combination of custom and common options. Legacy should not exist if somebody populates the legacy options we use those and we do not make them interact with the new config version. Merging legacy and new could give unpredictable results, at least this way it would be more predictable.

Then override are applied on top of this since they are chosen and forced by the application in code (which means they should never be overridden).